apps/client/src/common/components/view-params-editor/constants.ts

```diff
@@ -1,106 +1,122 @@
 import { UserFields } from 'ontime-types';
+import { TimeFormat } from 'ontime-types/src/definitions/core/TimeFormat.type';
 
 import { ParamField } from './types';
 
-export const TIME_FORMAT_OPTION: ParamField = {
+export const getTimeOption = (timeFormat: TimeFormat): ParamField => ({
   id: 'format',
   title: '12  / 24 hour timer',
   description: 'Whether to show the time in 12 or 24 hour mode. Overrides the global setting from preferences',
   type: 'option',
   values: { '12': '12 hour AM/PM', '24': '24 hour' },
-};
+  defaultValue: timeFormat,
+});
 
-export const CLOCK_OPTIONS: ParamField[] = [
-  TIME_FORMAT_OPTION,
+export const getClockOptions = (timeFormat: TimeFormat): ParamField[] => [
```

Comment:
is there a way we can avoid the extra step of having to find the times in every view?
we did that for the time string to abstract this into the central formatTime function. maybe here we can do something similar? perhaps exporting a closure that already has the timeFormat and just expects the field definition? (or` -insert-your-better-idea-here-`)

Reply:
It would probably need to be a hook: `useClockOptions`, but that's a chain reaction to rewriting other constants declarations.

Reply:
I say let's stick with this verbose solution (unless you really don't like it). I'm sure we'll revisit this again.

Reply:
Could the data be exposed from viewWrapper? It might already be

Reply:
Resolving in favor of newer discussions